Same commit as before, but works without errors/spam.

DraksterAU opened the pull request:
This release adds SQL as a dependency that is tested on startup, causing
the plugin to fail to start unless this is configured and connects, this
prevents log spam.

Also, added basic debug-mode config setting to hide stack traces and
only show english-ish errors for normal players/server admins. Should
avoid some of the posts seen on bukkitdev.

Reverted shortening of console message prefixes, StockMarket stands out
a lot better than SM.

Tested with the following:

Bukkit 1.8.8 / Vault v1.5.6b49 / CraftConomy3 v3.3.0
Bukkit 1.7.10 / Vault 1.4.1 / CraftConomy3 v3.2.1
KCauldron 1.7.10 / Vault 1.4.1 / CraftConomy3 v3.2.1

Commit:
Added plugin yml files

Commit:
v.08 Release

This release adds SQL as a dependency that is tested on startup, causing
the plugin to fail to start unless this is configured and connects, this
prevents log spam.

Also, added basic debug-mode config setting to hide stack traces and
only show english-ish errors for normal players/server admins. Should
avoid some of the posts seen on bukkitdev.

Reverted shortening of console message prefixes, StockMarket stands out
a lot better than SM.

Works with the following:

Bukkit 1.8.8 / Vault v1.5.6b49 / CraftConomy3 v3.3.0
Bukkit 1.7.10 / Vault 1.4.1 / CraftConomy3 v3.2.1
KCauldron 1.7.10 / Vault 1.4.1 / CraftConomy3 v3.2.1

NOTE: Even though CraftConomy was the plugin tested against, please feel
free to use any vault-version-appropriate Economy plugin!

Commit:
Fixed two problematic errors, they should be errors after the first attempt!

Commit:
Stop spam full stop. Constantly trying to update databases isnt fun.

mashlol commented on com/github/mashlol/MySQL.java:
```diff
@@ -133,8 +168,11 @@ public void close() {
+                            if (StockMarket.debugMode == true) {
+                                e.printStackTrace();
+                            }	
+                            dbstatus = false;
```
Can we pull all of these out into a method?  It can take the custom string to print as an arg

DraksterAU replied:
You mean like this - sort of? (Syntax might not be 100%): 

	public void throwMsg($type, $msg, $fatal) {
                    log.$type("[StockMarket] " + msg);
                    if ($type == 'severe') {
	                    this.disablePlugin();
                    }
                    return;
	}

mashlol replied:
I was thinking more like:

```
public void logError(String errorMessage, Exception e) {
  System.out.println("[StockMarket] " + errorMessage);
  if (StockMarket.debugMode) {
    e.printStackTrace();
  }
}
```

Not sure if we want to disable the plugin or not, maybe we do.

DraksterAU replied:
That's pretty good, I was just trying to use Bukkit logging levels and cater for messages as well.
Up to you?

Sent from my iPhone

> On 2 Jan 2016, at 3:54 PM, Kevin Bedi <notifications@github.com> wrote:
> 
> In com/github/mashlol/MySQL.java:
> 
> > @@ -133,8 +168,11 @@ public void close() {
> >  		try {
> >  			con.close();
> >  		} catch (SQLException e) {
> > -			System.out.println("A bad call to mysql.close() occurred");	
> > -			e.printStackTrace();
> > +                            System.out.println("[StockMarket] " + "General Database error. Could not close SQL connection after use! Enable debug-mode for more info.");
> > +                            if (StockMarket.debugMode == true) {
> > +                                e.printStackTrace();
> > +                            }	
> > +                            dbstatus = false;
> I was thinking more like:
> 
> public void logError(String errorMessage, Exception e) {
>   System.out.println("[StockMarket] " + errorMessage);
>   if (StockMarket.debugMode) {
>     e.printStackTrace();
>   }
> }
> Not sure if we want to disable the plugin or not, maybe we do.
> 
> —
> Reply to this email directly or view it on GitHub.
>

mashlol replied:
If you find a good way to do it, feel free.  It should be fine to just have all of them log at severe

mashlol commented on com/github/mashlol/StockMarket.java:
```diff
@@ -49,23 +52,27 @@ public void onDisable() {
-
+        public void disablePlugin() {
+			Bukkit.getServer().getPluginManager().disablePlugin(this);
+			return;
```
Rm the return

mashlol commented on com/github/mashlol/StockMarket.java:
```diff
@@ -100,6 +114,7 @@ public void loadConfiguration() {
 		maxPerPlayerPerStock = getConfig().getInt("max-total-stocks-per-player-per-stock");
 		
 		payOffline = getConfig().getBoolean("pay-offline-players");
+                debugMode = getConfig().getBoolean("debug-mode");
```
There's an extra space at the start of the line here

mashlol commented on com/github/mashlol/MySQL.java:
```diff
@@ -37,14 +37,22 @@ public MySQL () {
 			
 			//e.printStackTrace();
 		} catch (ClassNotFoundException e) {
 			// TODO Auto-generated catch block
```
You can remove these TODOs now that they do something :+1:

DraksterAU commented:
Agreed with your comments, ill get onto it shortly :)

Commit:
Disable plugin doesnt need to return

Commit:
Should be a decent method, does not handle ClassNotFoundExceptions though

Note, I've excluded the alter table statements as they need to be moved
ton onEnable first, or we get log spam.

Commit:
Removed TODO

Commit:
Removed unnecessary spacing.

DraksterAU commented:
@mashlol Should be good now bossman! ;)
